util/qartod_service.py

```diff
@@ -0,0 +1,123 @@
+import json
+import engine
+import logging
+import requests
+from copy import deepcopy
+
+log = logging.getLogger(__name__)
+
+
+class QartodTestServiceAPI(object):
+    def __init__(self, qartod_url):
+        self.base_url = qartod_url
+
+    def find_qartod_tests(self, subsite, node, sensor, stream, parameter):
+        result = []
+
+        params = {
+            'subsite': subsite,
+            'node': node,
+            'sensor': sensor,
+            'stream': stream,
+            'parameter': parameter
+        }
+
+        url = '/'.join((self.base_url, 'find'))
+        response = requests.get(url, params=params)
+        try:
+            payload = response.json()
+        except ValueError:
+            log.error('Error fetching QARTOD tests: %s', ValueError.message)
+            return result
+
+        if response.status_code == requests.codes.ok:
+            for record in payload:
+                if record.pop('@class', None) == '.QartodTestRecord':
+                    qartod_test_record = QartodTestRecord(**record)
+                    expanded_test_records = self.expand_qartod_record(qartod_test_record, subsite, node, sensor,
+                                                                      stream, parameter)
+                    result.extend(expanded_test_records)
+        else:
```

Comment:
could rewrite to check for the error and return an empty list (as the exception does above) and remove the unnecessary branch. This would also delay declaring `results` until it is used, making the code more readable:

```python
        if response.status_code == requests.codes.ok:
            log.error('Error fetching QARTOD tests: <%r> %r', response.status_code, payload)
            return []

        result = []
        for record in payload:
            if record.pop('@class', None) == '.QartodTestRecord':
                qartod_test_record = QartodTestRecord(**record)
                expanded_test_records = self.expand_qartod_record(qartod_test_record, subsite, node, sensor,
                                                                  stream, parameter)
                result.extend(expanded_test_records)
        return result
```